### PR TITLE
Add getPassword, setPassword, and deletePassword APIs

### PR DESCRIPTION
**Pull request by RMacfarlane**
Adopts vscode-encrypt and introduces new extension APIs.

### CODE REVIEW

File: src/vs/platform/encryption/electron-main/encryptionMainService.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { ICommonEncryptionService } from 'vs/platform/encryption/electron-main/common/encryptionService';
+import { encrypt, decrypt } from 'vscode-encrypt';
```
Comment (bpasero) [inline src/vs/platform/encryption/electron-main/encryptionMainService.ts:8]: @RMacfarlane to not block startup can you do an `await import('vscode-encrypt')` when needed? Any module we add to the startup will make startup slower.

File: src/vs/workbench/services/encryption/electron-browser/encryptionService.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (bpasero) [inline src/vs/workbench/services/encryption/electron-browser/encryptionService.ts:1]: @RMacfarlane why can this not move into `electron-sandbox` layer and be there for sandbox too?

**Review by bpasero — CHANGES_REQUESTED** [review]

File: extensions/github-authentication/src/common/keychain.ts [hunk 3]
```diff
@@ -59,23 +47,42 @@ export class Keychain {
...
+	async tryMigrate(): Promise<string | null | undefined> {
+		try {
+			const keytar = getKeytar();
+			if (!keytar) {
+				throw new Error('keytar unavailable');
+			}
+
+			const oldValue = await keytar.getPassword(`${vscode.env.uriScheme}-github.login`, 'account');
+			if (oldValue) {
+				await this.setToken(oldValue);
```
[the remark this answers is not in the document]
Comment (RMacfarlane) [reply extensions/github-authentication/src/common/keychain.ts:77]: Yes, thank you!

[the remark this answers is not in the document]
Comment (gjsjohnmurray) [reply extensions/github-authentication/src/common/keychain.ts:77]: I'm sure you know what you are doing, but I was surprised not to see you do the cleanup using keytar.deletePassword

File: src/vs/platform/encryption/electron-main/encryptionMainService.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/platform/encryption/electron-main/encryptionMainService.ts]: I am seeing `await require('vscode-encrypt')` but I think it should be `await import('vscode-encrypt')`

**Review by sbatten — CHANGES_REQUESTED** [review]
Test Review. Will Re-Request.

**Pull request merged**